### ISSUE TITLE
folders arent created in vivaldi browser when downloading a file

### ISSUE THREAD

**Issue by black-lug1a** [Simon-Tesla/RaccoonyWebEx#128]
in vivaldi files are just dumped straight into the downloads folder without being sorted. brave and firefox work just fine.

**Comment by Simon-Tesla** [comment]
Unfortunately this would be a bug on Vivaldi's side as Raccoony uses the API as documented. I'll leave the bug open to remind me to look for/file a bug on them but unfortunately I don't believe there's anything I can do on this end.

**Comment by Eupeptic1** [comment]
### tl;dr:

Works for me with the out-of-the-box settings on Vivaldi 5.2.2623.46 and stock Raccoony 1.2.5, as installed from the Chrome web store, on desktop Linux. I also tried to induce this bug a few different ways, but I couldn't reproduce.

@black-lug1a What version of Vivaldi do you have?  Are you using Vivaldi on Windows, Linux, OS X, or Android?  Like Simon-Tesla said, this is probably a bug in Vivaldi, but knowing that information will help us ~~either find the existing bug report for Vivaldi or~~ file a new bug report with Vivaldi.

### Long version:

I downloaded the latest version of Vivaldi for Linux from vivaldi.com as a `.deb`, and installed it in the same Debian 10 x86_64 VM that I use to test Raccoony in the newest Chrome and Firefox releases.  Vivaldi's About page says I have `5.2.2623.46 (Stable channel) stable (64-bit)`.

On the first start of Vivaldi, I took the middle of the three presets (it offers something like "minimal", "normal", "max", with progressively more features enabled) and the default on everything else.  I then went to the Chrome web store and installed Raccoony 1.2.5.  I didn't change any of Raccoony's default settings.

#### Stock settings

I went to e621's main page, didn't log in, ran a search, accepted the disclaimer, got a page of images, picked one, and clicked on it.  The image page loaded in my browser and the raccoon head appeared.  I clicked on the head, then clicked the "Download" button, and the image downloaded.  On the local filesystem, the image was saved in the directory I expected it to be, `~/Downloads/raccoony/e621/artist_name/`.

I repeated this test for one image each on FA and Weasyl, and it worked there too -  the images were saved in `~/Downloads/raccoony/furaffinity/artist_name/` and `~/Downloads/raccoony/weasyl/artist_name/`respectively.

I also went to Wikipedia and used the context-menu "Download with Raccoony" link on one image there. The image was saved in `~/Downloads/raccoony/en.wikipedia.org/`, as expected.

#### Induction attempt no. 1 - tilde in directory name

I was thinking about issue #90 and wondering what would happen in a Chromium-based browser if you tried to put a tilde in a *directory* name.  I was on Weasyl, so I changed the "Download path" setting in Raccoony from the stock `raccoony/{siteName}/{author}/{submissionId}_{filename}_by_{author}.{extension}` to `~/raccoony/{siteName}/{author}/{submissionId}_{filename}_by_{author}.{extension}`.  I then clicked on another image at Weasyl, got the raccoon-head menu, clicked on the head, and clicked the "Download" button.  On the local filesystem, the image was saved to `~/Downloads/_/raccoony/weasyl/artist_name/`, so that was OK.  In other words, the "path sanitize" code in Raccoony did the right thing.

#### Induction attempt no. 2 - directory permissions

I wondered if some kind of permissions problem on the local filesystem might cause this bug.  My Downloads folder was set 755, but I did `cd; chmod Downloads 655`.  I then attempted to download an image from FurAffinity with Raccoony.  The download started, but then Vivaldi popped up a file/directory picker window, set to `~/Downloads`.  I didn't change anything in the window and hit the `Save` button, and Vivaldi told me I didn't have access to that folder.  I changed the directory in the picker to just my home directory, hit the `Save` button, and the file got saved.

I was under the impression that it might be possible to use the POSIX ACL feature (on ext3) to allow a user to create a file but not a directory, but after a little quality time with the man page and Google, I'm not sure that's easily possible.

Google *did* tell me that NTFS can probably do this, but if an end user had set that up on their own machine, they'd probably know about it - it's several menus deep off of the properties menu in Explorer.  If the PC is under the care of an IT department or similar, I *think* it would be possible for that setting to be "baked in" to the image for the PC, or set remotely via the Windows management tools.  I would think it would be fairly rare for the IT department to *do* that, but I dunno.

#### Induction attempt no. 3 - Vivaldi download setting

I searched the Vivaldi forum and found [this post](https://forum.vivaldi.net/topic/35643/download-directory-does-not-work-properly?lang=en-US) which seemed somewhat related, but it was for Vivaldi 2.3 in March 2019.

In Vivaldi, `"V" menu > Settings > Downloads > Save Files to Default Location Without Asking` was unchecked, so I checked it, and then tried downloading a couple of images from Weasyl with Raccoony.  The setting did not affect Raccoony's behavoir; the images went to ~/Downloads/raccoony/weasyl/artist_name/` as expected.

#### Vivaldi bug search

Vivaldi's bug tracker is apparently a secret; see [this Vivaldi forum post](https://forum.vivaldi.net/topic/21605/where-is-the-bug-tracker) or [this Reddit post](https://www.reddit.com/r/vivaldibrowser/comments/obpnh3/how_to_access_vivaldis_issue_tracker/).  I Googled a few terms around this issue but I couldn't find anything that seemed to be exactly this bug.

**Comment by Simon-Tesla** [comment]
@black-lug1a If you can provide more details around this issue, please feel free to reopen this bug with them. Since we can't reproduce the issue, I'm going to go ahead and close this now.
